views/js/runner/plugins/tools/textToSpeech/textToSpeech.js

```diff
@@ -0,0 +1,374 @@
+            },
+
+            /**
+             * Click to pronounce
+             */
+            clickToSpeak: function clickToSpeak() {
+                options.enableClickToSpeak = !options.enableClickToSpeak;
+                this._exec('enableClickToSpeak', options.enableClickToSpeak);
+                this._exec('enableContinuousReading', !options.enableClickToSpeak); // continuous reading is off when click to speak is on
+                this.trigger('clickToSpeak');
+            }
+        }, {
+            // defaults
+        })
+        .setTemplate(tpl)
+        .on('init', function () {
+            var self = this;
+
+            speed = 40; // default speed
+            volume = 40; // default volume
+
+            // we have to mark some blocks as ignored to prevent TTS accessing it
+            $(options.ignoreEls.join(',')).each(function () {
+                $(this).attr('ignore', true);
+            });
+
+            require([options.toolbarUrl], function () {
+                var tss = self._get('texthelpSpeechStream');
+
+                require(["//" + tss.g_strServer + "/SpeechStream/v" + tss.g_strBuild + "/texthelpMain.js"], function () {
+                    self._init();
+                });
+            });
+        })
+        .init(config)
+        .on('render', function () {
+            var $this = this.getElement();
+
+            // prevents clicks from removing highlighted text
+            $this.on('mousedown', function (e) {
+                e.preventDefault();
+                return false;
+            });
+
+            // Action clicks
+            $this.find('.action').on('click', function (e) {
+                // prevents disabled actions from being triggered
+                if ($(this).hasClass('disabled')) {
+                    e.stopImmediatePropagation();
+                }
+
+                // hides settings menu when another action clicked
+                if (!$(this).closest('.settings').length) {
+                    $this.find('.settings').removeClass('active');
+                    $this.find('.settings > .settings-menu').hide();
+                }
+            });
+
+            // Show/hide settings menu
+            $this.find('.settings').on('click', function () {
+                $(this).toggleClass('active');
+                $this.find('.settings-menu').toggle();
+            });
+
+            // Settings menu
+            $this.find('.settings > .settings-menu')
+            .on('click', function (e) { // prevent child elements triggering a click on settings menu
+                if ($(this).closest('.settings-menu').length) {
+                    e.stopPropagation();
+                }
+            })
+            .hide(); // Hide settings menu to begin
+
+            // Hide/show volume and speed sliders
+            $this.find('.settings > .settings-menu > .option')
+            .on('hover', function () {
+                $(this).find('.slider-container').show();
```

Comment:
For this line and all below: you could use `ui/hider`for show/hide operation. It allow a centralized control on visibility of elements in TAO.